### PR TITLE
ocamlPackages.caqti: init at 1.3.0

### PR DESCRIPTION
**Pull request by bcc32**
###### Motivation for this change

Package the OCaml caqti library, a library for monadic cooperative-threaded access to various relational databases.  I added myself as maintainer, although I don't have much experience maintaining such packages.  If the relevant OCaml package reviewers would be willing/prefer to maintain these packages, I'm happy to accommodate.

I also tested these packages by building some personal projects that depend on caqti (https://github.com/bcc32/watch-later), so I can validate that at least `caqti-async`+`caqti-driver-sqlite3` works fine.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -1036,6 +1036,12 @@
     githubId = 1015044;
     name = "Brandon Carrell";
   };
+  bcc32 = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:1039]: Please put this into a separate commit with the message ``maintainers: add bcc32``.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/top-level/ocaml-packages.nix [hunk 2]
```diff
@@ -751,6 +751,8 @@ let
 
     pgocaml_ppx = callPackage ../development/ocaml-modules/pgocaml/ppx.nix {};
 
+    ocaml_postgresql = callPackage ../development/ocaml-modules/postgresql {};
```
Comment (vbgl) [inline pkgs/top-level/ocaml-packages.nix:772]: Please use `postgresql` as attribute name (i.e., the same as the name of the directory and the same as the name used by `dune` and `ocamlfind`). There are other ways to avoid infinite recursion than using a surprising name; see `gnuplot` for instance.

File: pkgs/development/ocaml-modules/postgresql/default.nix [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    repo = "postgresql-ocaml";
+    rev = version;
+    sha256 = "0fd96qqwkwjhv6pawk4wivwncszkif0sq05f0g5gd28jzwrsvpqr";
+  };
+
+  buildInputs = [ postgresql ];
+
+  meta = {
+    description = "Bindings to the PostgreSQL library";
+    license = "LGPL-2.1-or-later WITH OCaml-LGPL-linking-exception";
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/postgresql/default.nix:20]: Many OCaml derivations in `nixpkgs` just use `lib.licenses.lgpl21Plus` and do not worry about the linking exception.

**Review by vbgl — CHANGES_REQUESTED** [review]

**Comment by bcc32** [comment]
Thanks for the review.  I fixed up the comments, plus made some corrections to `ocamlPackages.mariadb`.

**Review by vbgl — APPROVED** [review]

**Comment by vbgl** [comment]
Merged as 1b6764da4e600e66c896a6746ceaa038a40a8f2b
